src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -291,7 +291,7 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
         {
             try
             {
-                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: project.Documents.Any(d => d.IsOpen())))
+                if (!skipClosedFileChecks && !CheckOption(project.Solution.Workspace, project.Language, documentOpened: false))
```

Comment:
by the way, in v1 engine, only compilation (end) action can produce diagnostics on other document. it is explicitly blocked in v1. it will be v2 new feature.